lib/matplotlib/type1font.py

```diff
@@ -35,9 +35,292 @@
+
+    @classmethod
+    def _escape(cls, match):
+        group = match.group(1)
+        try:
+            return cls._replacements[group]
+        except KeyError:
+            return chr(int(group, 8))
+
+    @functools.lru_cache()
+    def value(self):
+        if self.raw[0] == '(':
+            return self._escapes_re.sub(self._escape, self.raw[1:-1])
+        else:
+            data = self._ws_re.sub('', self.raw[1:-1])
+            if len(data) % 2 == 1:
+                data += '0'
+            return binascii.unhexlify(data)
+
+
+class _BinaryToken(_Token):
+    kind = 'binary'
+
+    def value(self):
+        return self.raw[1:]
+
+
+class _NumberToken(_Token):
+    kind = 'number'
+
+    def is_number(self):
+        return True
+
+    def value(self):
+        if '.' not in self.raw:
+            return int(self.raw)
+        else:
+            return float(self.raw)
+
+
+def _tokenize(data: bytes, skip_ws: bool):
+    """
+    A generator that produces _Token instances from Type-1 font code.
+
+    The consumer of the generator may send an integer to the tokenizer
+    to indicate that the next token should be _BinaryToken of the given
+    length.
+
+    Parameters
+    ----------
+    data : bytes
+        The data of the font to tokenize.
+
+    skip_ws : bool
+        If true, the generator will drop any _WhitespaceTokens from the output.
+    """
+
+    text = data.decode('ascii', 'replace')
+    whitespace_or_comment_re = re.compile(r'[\0\t\r\f\n ]+|%[^\r\n]*')
+    token_re = re.compile(r'/{0,2}[^]\0\t\r\f\n ()<>{}/%[]+')
+    instring_re = re.compile(r'[()\\]')
+    hex_re = re.compile(r'^<[0-9a-fA-F\0\t\r\f\n ]*>$')
+    oct_re = re.compile(r'[0-7]{1,3}')
+    pos = 0
+    next_binary = None
+
+    while pos < len(text):
+        if next_binary is not None:
+            n = next_binary
+            next_binary = (yield _BinaryToken(pos, data[pos:pos+n]))
+            pos += n
+            continue
+        match = whitespace_or_comment_re.match(text, pos)
+        if match:
+            if not skip_ws:
+                next_binary = (yield _WhitespaceToken(pos, match.group()))
+            pos = match.end()
+        elif text[pos] == '(':
+            # PostScript string rules:
+            # - parentheses must be balanced
+            # - backslashes escape backslashes and parens
+            # - also codes \n\r\t\b\f and octal escapes are recognized
+            # - other backslashes do not escape anything
+            start = pos
+            pos += 1
+            depth = 1
+            while depth:
+                match = instring_re.search(text, pos)
+                if match is None:
+                    raise ValueError(
+                        f'Unterminated string starting at {start}')
+                pos = match.end()
+                if match.group() == '(':
+                    depth += 1
+                elif match.group() == ')':
+                    depth -= 1
+                else:  # a backslash
```

Comment:
I guess that here you don't really care about handling the backslash escapes, and all you want to do is simply to match the (unescaped) parentheses, so you could perhaps just replace instring_re by something like `(?<!\\)[()]` (parentheses not preceded by a backslash)?

Reply:
Ah, I gave this a try but it wouldn't work because the parenthesis *can* be preceded by backslashes if the backslash is itself escaped (i.e. one would really need to search for "parenthesis not preceded by an even number of backslashes").  So let's forget about this for now.